lib/inc/internal/unmergeable.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <hocon/config_value.hpp>
+#include <vector>
+
+namespace hocon {
+
+    /**
+    * Interface that tags a ConfigValue that is not mergeable until after
+    * substitutions are resolved. Basically these are special ConfigValue that
+    * never appear in a resolved tree, like {@link ConfigSubstitution} and
+    * {@link ConfigDelayedMerge}.
+    */
+    class unmergeable {
+    public:
+        virtual std::vector<shared_value> unmerged_values() = 0;
```

Comment:
The Java implementation does this:
```
Collection<? extends AbstractConfigValue> unmergedValues();
```

I *think* this means a collection of the type of any of `AbstractConfigValue`'s children, so is a vector of pointers to the base `config_value`s right here?